### PR TITLE
EnumAttributeConverter: enums can be identified by toString() or name()

### PR DESCRIPTION
**Pull request by martinKindall**
## Motivation and Context
https://github.com/aws/aws-sdk-java-v2/issues/3958

## Modifications
Created static method EnumAttributeConverter::createWithNameAsKeys which creates a converter based on the Enum::name method to identify enums, rather than Enum::toString. This is preferable because Enum::name is final and cannot be overwritten, as opposed to Enum::toString. EnumAttributeConverter::create is kept as it is, for backward compatibility.

## Testing
The included tests were tested running the unit tests of that module with ./mvnw package
Testing environment

Java version: 1.8.0_361, vendor: Oracle Corporation
OS name: "windows 10", version: "10.0", arch: "amd64"

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License
- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/EnumAttributeConverter.java [hunk 1]
```diff
@@ -41,23 +48,30 @@
...
         this.enumValueMap = Collections.unmodifiableMap(mutableEnumValueMap);
     }
 
     public static <T extends Enum<T>> EnumAttributeConverter<T> create(Class<T> enumClass) {
-        return new EnumAttributeConverter<>(enumClass);
+        return new EnumAttributeConverter<>(enumClass, Enum::toString);
+    }
+
+    public static <T extends Enum<T>> EnumAttributeConverter<T> createWithNameAsKeys(Class<T> enumClass) {
+        return new EnumAttributeConverter<>(enumClass, Enum::name);
```
Comment (rkennedy-mode) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/EnumAttributeConverter.java:69]: I think this means that the default converter that virtually everyone will use (by virtue of using DefaultAttributeConverterProvider) will continue using the unsafe toString() version. That's fine if that's the way you all want to go, but I figured I'd point it out.

Comment (martinKindall) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/EnumAttributeConverter.java:69]: Exactly, this is for backward compatibility.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/EnumAttributeConverter.java [hunk 2]
```diff
@@ -41,23 +48,30 @@
...
 
         this.enumValueMap = Collections.unmodifiableMap(mutableEnumValueMap);
     }
 
     public static <T extends Enum<T>> EnumAttributeConverter<T> create(Class<T> enumClass) {
-        return new EnumAttributeConverter<>(enumClass);
+        return new EnumAttributeConverter<>(enumClass, Enum::toString);
+    }
+
+    public static <T extends Enum<T>> EnumAttributeConverter<T> createWithNameAsKeys(Class<T> enumClass) {
```
[the remark this answers is not in the document]
Comment (martinKindall) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/EnumAttributeConverter.java]: Done.

**Review by zoewangg — APPROVED** [review]
Thank you again for your contribution!

**Comment by zoewangg** [comment]
Running the builds now

**Pull request merged**